mkdocs/contrib/search/__init__.py

```diff
@@ -67,7 +68,7 @@ def on_config(self, config: MkDocsConfig, **kwargs) -> MkDocsConfig:
             path = os.path.join(base_path, 'templates')
             config.theme.dirs.append(path)
             if 'search/main.js' not in config.extra_javascript:
-                config.extra_javascript.append('search/main.js')
+                config.extra_javascript.append('search/main.js')  # type: ignore
```

Comment:
I'm wondering – is there no possibility to define something like `str | ExtraScriptValue` in the validation/config system? Probably something like `ListOfItems((str, ExtraScriptValue))`, or in any other way that is Python 3.7 compatible?

Reply:
I tried this in many ways but it makes mypy very confused. `ListOfItems` needs to contain a `SubConfig`, not any arbitrary Python type.
Adding a string will keep working but will generate a type error, sadly.